bundle/edu.gemini.pot/src/main/java/edu/gemini/pot/sp/memImpl/MemObsExecLog.java

```diff
@@ -16,11 +27,68 @@
         program = prog;
     }
 
-    @Override public final SPComponentType getType() { return OBS_EXEC_LOG; }
+    @Override public SPComponentType getType() { return OBS_EXEC_LOG; }
 
-    @Override public final void accept(ISPProgramVisitor visitor) {
+    @Override public void accept(ISPProgramVisitor visitor) {
         visitor.visitObsExecLog(this);
     }
 
     @Override public MemProgram getProgram() { return program; }
+
+    // Start REL-3986 ------ logging to debug an issue with missing events
+    private String formatEvents(ObsExecLog log) {
+        final StringBuilder buf = new StringBuilder();
+        log.getRecord().getAllEventList().forEach(e -> {
+            if ((e instanceof StartVisitEvent) || (e instanceof EndVisitEvent)) {
+                buf.append(String.format("\t%s\n", e));
+            }
+        });
+        return buf.toString();
+    }
+
+    private Option<String> formatEvents(String name, Object obj) {
+
+        return (DATA_OBJECT_KEY.equals(name) && (obj instanceof ObsExecLog)) ?
+                new Some<>(formatEvents((ObsExecLog) obj))                   :
+                ImOption.empty();
+
+    }
+
+    private Option<String> getObsId() {
+        return ImOption.apply(getContextObservation())
+                       .flatMap(o -> ImOption.apply(o.getObservationID()))
+                       .map(SPObservationID::stringValue);
+    }
+
+    @Override public PropagationId putClientData(String name, Object obj) {
```

Comment:
Whenever the "shell" containing observation execution log data is updated, we'll dump some event information out.  A theory is that start/end visits are recorded correctly but subsequently get overwritten (perhaps by sync code?).  This logging should at least answer that question.